src/app/components/show-user/show-user.component.ts

```diff
@@ -8,9 +8,19 @@ import { DataStoreService } from '../../services/data-store.service';
 })
 export class ShowUserComponent implements OnInit {
 
+  public checkUserExist = false;
   constructor(
-    private store: DataStoreService
-  ) { }
+    private store: DataStore
+  ) {
+    store.users.subscribe(
```

Comment:
Because of this check there are two different subscriptions happening to `store.users`. Once here and once in the template using `async` pipe. We don't want multiple subscriptions.

Reply:
so, we can probably use the subscription in the typescript file and the returned object from the subscription can be used by the template